crates/lang/macro/tests/unique_topics.rs

```diff
@@ -0,0 +1,98 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract]
+mod my_contract {
+    #[ink(storage)]
+    pub struct MyContract {}
+
+    /// Exemplary event
+    #[ink(event)]
+    pub struct MyEvent {
+        #[ink(topic)]
+        v0: Option<AccountId>,
+        #[ink(topic)]
+        v1: Balance,
+        #[ink(topic)]
+        v2: u128,
+        #[ink(topic)]
+        v3: bool,
```

Comment:
Maybe introduce some other field with literally the same type and emit value to be extra sure.